### DISCUSSION TITLE
My Blazor WASM runtime is currently not supported by QuestPDF

### DISCUSSION THREAD

**Discussion by troika-support** [QuestPDF/QuestPDF#931] [Q&A] [answered]
I am running a Blazor ASPNET Core Hosted application with PDF generation capabilities. I was able to generate PDFs before within my app while it was running on .NET 7. On upgrading it to .NET 8 and the QuestPDF package it started showing me the following error when trying to run code that generates PDF:

'The QuestPDF library has encountered an issue while loading one of its dependencies.\n\nYour runtime is currently not supported by QuestPDF. Currently supported runtimes are: win-x86, win-x64, linux-x64, linux-arm64, linux-musl-x64, osx-x64, osx-arm64. \n\nYour current runtime is detected as 'browser-wasm'.'

What should I do to fix this?

**Comment by MarcinZiabek** [comment, marked as answer]
Thank you for your message 😄 

For now, please continue using the `2023.12.X` version. It supports WASM and is still actively maintained.

The QuestPDF 2024 release introduced a significant internal redesign. While providing many exciting features and capabilities, it also temporarily dropped support for specific platforms. Based on Community feedback, we are reintroducing support for various platforms. 

I am happy to share that WASM is already a work in progress.

**Reply by troika-support** [reply]
Thank you for the feedback, it worked.

Only issue I have now is making it work in Safari browser.

**Reply by HClausing** [reply]
> Thank you for your message 😄
> 
> For now, please continue using the `2023.12.X` version. It supports WASM and is still actively maintained.
> 
> The QuestPDF 2024 release introduced a significant internal redesign. While providing many exciting features and capabilities, it also temporarily dropped support for specific platforms. Based on Community feedback, we are reintroducing support for various platforms.
> 
> I am happy to share that WASM is already a work in progress.

I'm happy to know about the `work in progress` state for WASM in 2024. Please keep going!

Thanks!

**Reply by MarcinZiabek** [reply]
@troika-support @HClausing Could you please share your thoughts regarding https://github.com/QuestPDF/QuestPDF/discussions/622#discussioncomment-10735147? Thank you 😄

**Reply by HClausing** [reply]
> @troika-support @HClausing Could you please share your thoughts regarding [#622 (comment)](https://github.com/QuestPDF/QuestPDF/discussions/622#discussioncomment-10735147)? Thank you 😄

For sure! Thanks!

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
